src/iosrv/init.c

```diff
@@ -589,10 +590,57 @@ server_init(int argc, char *argv[])
 	return rc;
 }
 
+static int
+cleanup_cb(const char *path, const struct stat *sb, int flag,
+           struct FTW *ftwbuf)
+{
+        int rc;
+
+        if (ftwbuf->level == 0)
+                return 0;
+
+        if (flag == FTW_DP || flag == FTW_D)
+                rc = rmdir(path);
+        else
+                rc = unlink(path);
+        if (rc)
+                D_ERROR("failed to remove %s\n", path);
+        return rc;
+}
+
+static void
+shared_memory_file_cleanup(void)
+{
+        int rc;
+
+        if (dss_nvme_shm_id != DAOS_NVME_SHMID_NONE) {
+                /* unlink spdk shared memory files for this stream */
+                char buffer[NAME_MAX];
+
+		/* remove shared memory dir created byi each daos io searver
+		 * currently in /var/run/dpdk
+		 */
+                snprintf(buffer, NAME_MAX, "/var/run/dpdk/spdk%d",
+			 dss_nvme_shm_id);
+                rc = nftw(buffer, cleanup_cb, 1, FTW_D);
+                if (rc == 0)
```

Comment:
(style)  code indent should use tabs where possible
(style)  suspect code indent for conditional statements (16, 25)